images/kindnetd/cmd/kindnetd/routes.go

```diff
@@ -24,30 +24,49 @@ import (
 	"k8s.io/klog/v2"
 )
 
-func syncRoute(nodeIP string, podCIDRs []string) error {
+func syncRoutes(nodeIP string, podCIDRs []string) error {
 	ip := net.ParseIP(nodeIP)
-
 	for _, podCIDR := range podCIDRs {
-		// parse subnet
-		dst, err := netlink.ParseIPNet(podCIDR)
-		if err != nil {
+		if err := syncRoute(podCIDR, ip); err != nil {
 			return err
 		}
+	}
+	return nil
+}
 
-		// Check if the route exists to the other node's PodCIDR
-		routeToDst := netlink.Route{Dst: dst, Gw: ip}
-		route, err := netlink.RouteListFiltered(nl.GetIPFamily(ip), &routeToDst, netlink.RT_FILTER_DST)
-		if err != nil {
-			return err
+func syncRoute(podCIDR string, gw net.IP) error {
+	// parse subnet
+	dst, err := netlink.ParseIPNet(podCIDR)
+	if err != nil {
+		return err
+	}
+
+	// List all routes to the other node's PodCIDR
+	routeToDst := netlink.Route{Dst: dst, Gw: gw}
+	routes, err := netlink.RouteListFiltered(nl.GetIPFamily(gw), &routeToDst, netlink.RT_FILTER_DST)
+	if err != nil {
+		return err
+	}
+
+	// Check if the route exists to the other node's PodCIDR and delete invalid routes
+	for _, route := range routes {
+		// Skip creation of a route if there is gateway which matches the expected.
+		if route.Gw.Equal(gw) {
+			return nil
```

Comment:
https://github.com/kubernetes-sigs/kind/pull/2941/files#r980086210

```suggestion
			continue
```

Reply:
hmm, we have to add some boolean to indicate that we found it, to avoid to create it after the loop